ecl/hqlcpp/hqlcpp.cpp

```diff
@@ -7487,6 +7487,83 @@ void HqlCppTranslator::doBuildAssignExecuteWhen(BuildCtx & ctx, const CHqlBoundT
     }
 }
 
+void HqlCppTranslator::doBuildExprPow(BuildCtx & ctx, IHqlExpression * expr, CHqlBoundExpr & tgt)
+{
+    assertex(expr->numChildren() == 2);
+    
+    IHqlExpression * left = expr->queryChild(0);
+    IHqlExpression * right = expr->queryChild(1);
+
+    bool notIntegerPower = !right->queryType()->isInteger();
+    if (notIntegerPower)
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    IValue * value = right->queryValue();
+    bool notConstant = (value == nullptr);
+    if (notConstant) 
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    const uint MAX_EXP_INLINE_VALUE = 3;
+    int64_t rightIntVal = value->getIntValue();
+    if (abs(rightIntVal) > MAX_EXP_INLINE_VALUE)
+    {
+        doBuildExprSysFunc(ctx, expr, tgt, powerId);
+        return;
+    }
+
+    IHqlExpression* outExpr = nullptr;
+    if (rightIntVal == 0)
+    {
+        outExpr = createConstant(left->queryType()->castFrom(1.0));
+    }
+    else if (rightIntVal == 1)
+    {
+        outExpr = left;
+    }
+    else
+    {
+        //---------------------------------------------------------------------------
+        // Question:
+        // Is there a better way to get a temp real here?
+        //---------------------------------------------------------------------------
+
+        CHqlBoundExpr leftBound;
+        buildTempExpr(ctx, left, leftBound);
+
+        CHqlBoundExpr tempReal;
+        OwnedITypeInfo realType = makeRealType(DEFAULT_REAL_SIZE);
+        doBuildCastViaTemp(ctx, realType, leftBound, tempReal);
+        outExpr = tempReal.expr;
+
+        for (int i = 1; i < abs(rightIntVal); i++)
+        {
+            outExpr = createValue(no_mul, realType, LINK(outExpr), LINK(tempReal.expr));
+        }
+
+        if (rightIntVal < 0)
+        {
+            CHqlBoundExpr bound;
+            buildTempExpr(ctx, outExpr, bound);
+
+            //---------------------------------------------------------------------------
+            // Question:
+            // This inserts a check for divide by zero. What is the best way to not have
+            // that check generated?
```

Comment:
The value could be zero, so it needs to have that check.

Reply:
Right, I was looking to move that check above the entire expansion. I ended up doing this during the code review changes, not sure if the approach I took was the best route though.